File: baekjoon/gold/1011/1011.cpp
#include <iostream>
#include <math.h>
int main() {
  int t;
  float x, y, r, distance;

  scanf("%d", &t);
  for (int i = 0; i < t; i++) {
    scanf("%f %f", &x, &y);
    distance = y - x;
    if (distance == 1) {
      printf("1\n");
    } else if (distance == 2) {
      printf("2\n");
    } else {
      r = floor(sqrt(distance));
      if ((r + 1) * r < distance) {
        r += 1.0;
      }
      int r2 = static_cast<int>(r);
      if (distance <= r * r) {
        printf("%d\n", 2 * r2 - 1);
      } else {
        printf("%d\n", 2 * r2);
      }
    }
  }
}

File: baekjoon/gold/1011/1011.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
)

var (
	reader *bufio.Reader = bufio.NewReader(os.Stdin)
	writer *bufio.Writer = bufio.NewWriter(os.Stdout)
)

func main() {
	var t, x, y, r, distance int

	defer writer.Flush()
	fmt.Fscanf(reader, "%d\n", &t)
	for i := 0; i < t; i++ {
		fmt.Fscanf(reader, "%d %d\n", &x, &y)
		distance = y - x
		if distance == 1 {
			fmt.Fprintln(writer, 1)
		} else if distance == 2 {
			fmt.Fprintln(writer, 2)
		} else {
			r = int(math.Sqrt(float64(distance)))
			if (r+1)*r < distance {
				r++
			}
			if distance <= r*r {
				fmt.Fprintln(writer, r*2-1)
			} else {
				fmt.Fprintln(writer, r*2)
			}
		}
	}
}
